straxen/bokeh_utils.py

```diff
@@ -0,0 +1,134 @@
+        to plot peaks.
+    """
+    if not (np.all(peaks['type'] == peaks[0]['type'])):
+        raise ValueError('All peaks must be of the same type (S1, S2 or Unknown)!')
+
+    x_p = []
+    y_p = []
+    for ind, p in enumerate(peaks):
+        x, y = _patches_x_y(p, keep_amplitude_per_sample=keep_amplitude_per_sample)
+        x -= relative_start  # relative to first peak
+        x = x / time_scaler
+        x_p.append(x)
+        y_p.append(y)
+
+    if peaks[0]['type'] == 2:
+        scaler = 10**-3
+    else:
+        scaler = 1
+
+    source = bklt.ColumnDataSource(data={'xs': x_p,  # Coordinates for peak patches
+                                         'ys': y_p,
+                                         'x': peaks['x'],  # XY-Pos in PMZ Hitpattern
+                                         'y': peaks['y'],
+                                         'time': peaks['time'],
+                                         'center_time': peaks['center_time'],
+                                         'endtime': strax.endtime(peaks),
+                                         'width_50': peaks['range_50p_area'] * scaler,
+                                         'width_90': peaks['range_90p_area'] * scaler,
+                                         'rise': peaks['rise_time'] * scaler,
+                                         'rel_center_time': (peaks['center_time'] - peaks['time']) * scaler,
+                                         'area': peaks['area'],
+                                         'aft': peaks['area_fraction_top'],
+                                         'nhits': peaks['n_hits'],
+                                         }
+                                   )
+    return source
+
+
+def _patches_x_y(peak, keep_amplitude_per_sample=False):
+    """
+    Creates x,y coordinates needed to draw peaks via
+    bokeh.models.patches.
+
+    :param peak: Peak for which we need the x/y samples
+    :param keep_amplitude_per_sample: If y-data should be in units of "per sample"
+        or "per ns".
+
+    :returns: Tuple of x, y
+    """
+    if keep_amplitude_per_sample:
+        dt_a = 1
+    else:
+        dt_a = peak['dt']
+
+    x = [0] + list(np.arange(peak['length'])) + [peak['length']]
+    x = np.array(x) * peak['dt'] + peak['time']
+    y = [0] + list(peak['data'][:peak['length']]) + [0]
+    y = np.array(y) / dt_a
+    return x, y
+
+
+def peak_tool_tip(peak_type):
+    """
+    Default mouseover tooltip for peaks.
+
+    :param peak_type: If 2, all time variables are in µs else in ns.
+    :return: dictionary of tooltips. Can be converted to a list for
+        bokeh.models.HoverTool.
+    """
+
+    # Add static time parameters:
+    tool_tip = {"time_static": ("time [ns]", "@time"),
+                "center_time": ("center_time [ns]", "@center_time"),
+                "endtime": ("endtime [ns]", "@endtime"),
+                }
+
+    # Now ns/µs parameters for S1 and S2
+    if peak_type == 2:
```

Comment:
could reduce copy paste by just doing `if peak_type !=2: for v in tool_tip.values(): v[0].replace('µ', 'n')`

Reply:
Hmm not replacing all "n"s but yes I can change it.

Reply:
Okay addressed it